processing/src/main/java/io/druid/segment/IndexIO.java

```diff
@@ -224,20 +225,26 @@ public static boolean convertSegment(File toConvert, File converted, IndexSpec i
             converted,
             indexSpec
         );
-        return true;
+        wasConverted = true;
+        break;
       case 8:
         DefaultIndexIOHandler.convertV8toV9(toConvert, converted, indexSpec);
-        return true;
+        wasConverted = true;
+        break;
       default:
         if (forceIfCurrent) {
           IndexMaker.convert(toConvert, converted, indexSpec);
-          DefaultIndexIOHandler.validateTwoSegments(toConvert, converted);
-          return true;
+          wasConverted = true;
```

Comment:
We used to only validate v8->v9 conversion. I don't see a reason to change this, since we don't know what the implications may be for older versions. It would also keep the code a lot more readable.